docs/book/src/reference/versions.md

```diff
@@ -78,6 +78,24 @@ The Kubeadm Control Plane Provider talks to the API server and etcd members of e
 
 The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 
+#### clusterctl
+
+It is strongly recommended using a version of [clusterctl](../clusterctl/overview.md) that exactly matches the version
+of the Cluster API core provider; in case of upgrades, clusterctl should be upgraded first and used to upgrade
+all the other components.
+
+<aside class="note warning">
+
+<h1>Warning!</h1>
+
+clusterctl was re-designed in the v1alpha3/v0.3 release series.
+
+A clusterctl version in the v1alpha3/v0.3 is expected to manage Cluster API providers the current and all the previous 
+patch releases in the same series; however, considering the project is still in alpha, we are not providing any guarantee
+for using an older version of clusterctl to install the newest patch releases. 
```

Comment:
I wonder if we need to mention the v1alpha3 in this statement here, we mention it on the line above and it's likely to be true for v1alpha4, v1alpha5 etc as well right? 

Perhaps:

```
Each clusterctl version is expected to manage Cluster API providers with a matching release as well as all previous patch releases in the same series (eg. v0.3.5 is compatible with v0.3.0 through v0.3.5); ...
```
I think this is what you are trying to say here? I'm a bit confused though, as the provider implementations have their own versioning schemes don't they? Eg Azure is on v0.4.x now even though that's compatible with cluster-api v0.3.x?

Reply:
You got this almost right, the trick here is that the release of clusterctl is linked mostly to the release of CAPI (all the other provider are linked indirectly to CAPI via the contract)

Reply:
Oh right, in which case, should we change the word `provider`? That makes me think of the providers, perhaps `components` or something is better to mean the core CAPI controllers?

Reply:
I have removed this note because now this is implicit in the previous statement